scipy/sparse/_construct.py

```diff
@@ -591,64 +595,69 @@ def kron(A, B, format=None):
     if isinstance(A, sparray) or isinstance(B, sparray):
         # convert to local variables
         bsr_sparse = bsr_array
-        csr_sparse = csr_array
         coo_sparse = coo_array
     else:  # use spmatrix
         bsr_sparse = bsr_matrix
-        csr_sparse = csr_matrix
         coo_sparse = coo_matrix
 
     B = coo_sparse(B)
-    if B.ndim != 2:
-        raise ValueError(f"kron requires 2D input arrays. `B` is {B.ndim}D.")
 
-    # B is fairly dense, use BSR
-    if (format is None or format == "bsr") and 2*B.nnz >= B.shape[0] * B.shape[1]:
-        A = csr_sparse(A,copy=True)
-        if A.ndim != 2:
-            raise ValueError(f"kron requires 2D input arrays. `A` is {A.ndim}D.")
-        output_shape = (A.shape[0]*B.shape[0], A.shape[1]*B.shape[1])
+    # B is 2D and fairly dense, use BSR
+    if (
+        (format is None or format == "bsr") and
+        B.ndim == 2 and 2*B.nnz >= math.prod(B.shape)
+    ):
+        A = coo_sparse(A, copy=True)
+        if A.ndim == 2:
+            A = A.tocsr()
```

Comment:
I'm a little confused here: why do we first copy A into COO format, then immediately convert to CSR?

And if A.ndim isn't 2, then we'll end up copying A again on line 626.